### PR TITLE
[Consensus] Fixed `TestPacemakerCatchupSameStepDifferentRounds` test

### PR DESCRIPTION
**Pull request by deblasis**
## Description

This PR removes the call to `SetLeaderId` via reflection and it seems that this fixes the test. 

Previously we were calling
```go
        consensusModImpl.MethodByName("SetLeaderId").Call([]reflect.Value{reflect.Zero(reflect.TypeOf(&leaderId))})
```

and the "Zero Value" of a pointer is `nil`:

https://go.dev/play/p/gml5ADTLNLZ

image

SetLeaderId` was:
```go
func (m *consensusModule) SetLeaderId(leaderId *typesCons.NodeId) {
	m.leaderId = leaderId
}
```

so we were essentially setting the leader to `nil`. 

Removing the `SetLeaderId` call altogether seems to have the same result so I now wonder if the test logic is correct.

## Issue

N/A (<4h of work)

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [x] Bug fix
- [ ] Code health or cleanup
- [ ] Major breaking change
- [ ] Documentation
- [ ] Other 

## List of changes

- Removed `SetLeaderId` function from the interface and test.

## Testing

- [x] `make develop_test`
- [x] [LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md) w/ all of the steps outlined in the `README`

## Required Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have tested my changes using the available tooling
- [x] I have updated the corresponding CHANGELOG

### If Applicable Checklist

- [ ] I have updated the corresponding README(s); local and/or global
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added, or updated, [mermaid.js](https://mermaid-js.github.io) diagrams in the corresponding README(s)
- [ ] I have added, or updated, documentation and [mermaid.js](https://mermaid-js.github.io) diagrams in `shared/docs/*` if I updated `shared/*`README(s)

### CODE REVIEW

File: consensus/state_sync/helpers.go [hunk 1]
```diff
@@ -34,9 +34,9 @@ func (m *stateSync) sendToPeer(msg *anypb.Any, peerId cryptoPocket.Address) erro
 func getMessageType(msg *typesCons.StateSyncMessage) string {
 	switch msg.Message.(type) {
 	case *typesCons.StateSyncMessage_MetadataReq:
-		return "MetadataRequest"
```
Comment (Olshansk) [inline consensus/state_sync/helpers.go:37]: This is how it's being used:

```go
	fields := map[string]any{
		"height":     height,
		"peerId":     peerId,
		"proto_type": getMessageType(stateSyncMsg),
	}

	m.logger.Info().Fields(fields).Msg("Sending StateSync Message")
	return m.sendToPeer(anyMsg, peerId)
```

I think we should leave this as `MetadataRequest` OR rename `MetadataRequest` to `PeerMetadataRequest`, but `StateSyncMetadataResponse` seems wrong.

**Review by Olshansk — CHANGES_REQUESTED** [review]
@gokutheengineer 

1. Sorry for merging in broken code - no excuses on my emd

2. I didn't realize that the investigation to fix the logs was going to take more time such that a broken CI was going to rollover onto the next day. Let's split hotfix back into its own PR and do the followup work separately. Let's always go with this approach moving forward

File: consensus/debugging.go [hunk 2]
```diff
@@ -194,7 +189,7 @@ func (m *consensusModule) sendGetMetadataStateSyncMessage(_ *messaging.DebugMess
 		}
 		valAddress := cryptoPocket.AddressFromString(val.GetAddress())
 		if err := m.stateSync.SendStateSyncMessage(stateSyncMetaDataReqMessage, valAddress, requestHeight); err != nil {
-			m.logger.Error().Err(err).Str("proto_type", "GetMetaDataRequest").Msg("failed to send StateSyncMessage")
+			m.logger.Error().Err(err).Str("proto_type", "StateSyncMetadataRequest").Msg("failed to send StateSyncMessage")
```
Comment (Olshansk) [inline consensus/debugging.go:192]: This isn't the actual proto type: `StateSyncMetadataRequest`.

File: consensus/state_sync/helpers.go [hunk 3]
```diff
@@ -34,13 +34,13 @@ func (m *stateSync) sendToPeer(msg *anypb.Any, peerId cryptoPocket.Address) erro
...
+		return "PeerMetadataRequest"
 	case *typesCons.StateSyncMessage_MetadataRes:
-		return "MetadataResponse"
+		return "PeerMetadataResponse"
```
Comment (Olshansk) [inline consensus/state_sync/helpers.go:39]: Let's be consistent with `StateSyncMetadataRequest` or `MetadataRequest` etc..

We're mixing things all over. What's your thinking?

**Review by Olshansk — CHANGES_REQUESTED** [review]

File: consensus/state_sync/helpers.go (hunk 3 above)
Comment (gokutheengineer) [reply consensus/state_sync/helpers.go:39]: I think it is best to update all to actual proto names, to remove confusion and mixing, so:
- StateSyncMetadataRequest
- StateSyncMetadataResponse 
- GetBlockRequest 
- GetBlockResponse

Wdyt?

File: consensus/debugging.go (hunk 2 above)
Comment (gokutheengineer) [reply consensus/debugging.go:192]: but it is: https://github.com/pokt-network/pocket/blob/main/consensus/types/proto/state_sync.proto

Comment (Olshansk) [reply consensus/debugging.go:192]: Good point

File: consensus/state_sync/helpers.go (hunk 3 above)
Comment (Olshansk) [reply consensus/state_sync/helpers.go]: Good idea, let's do it

**Review by gokutheengineer — APPROVED** [review]

**Review by Olshansk — APPROVED** [review]

**Pull request merged**